### PR TITLE
[Optimization] Use a separate driver when using Triton with Paddle

### PR DESCRIPTION
**Pull request by SigureMo**
## Motivation

> :bulb: If this PR is a Cherry Pick, the PR title needs to follow the format by adding the [Cherry-Pick] label at the very beginning and appending the original PR ID at the end. For example, [Cherry-Pick][CI] Add check trigger and logic(#5191)

> :bulb: 如若此PR是Cherry Pick，PR标题需遵循格式，在最开始加上[Cherry-Pick]标签，以及最后面加上原PR ID，例如[Cherry-Pick][CI] Add check trigger and logic(#5191)

解决在混合 torch 场景使用的 compat guard 过重导致性能下降的问题

## Modifications

将在 Paddle 下使用 triton kernel 的方式分为两种

1. 不混合 PyTorch 的场景，仍然直接开启 `enable_compat(scope={"triton"})`，后面和普通用法一样使用，如 https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/custom_op/cross_ecosystem_custom_op/user_guide_cn.html 中所述，零运行时开销
2. 在混合 PyTorch 的场景，为保证在运行 kernel 时所使用的 driver 里的 `torch` API 被替换为了 Paddle API，需要在 kernel 上加一个装饰器 `enable_compat_on_triton_kernel`

   ```python
   @enable_compat_on_triton_kernel
   @triton.jit
   def triton_kernel_xxx(a, b, c):
       ...
   ```

   这会在跑这个 triton kernel 前将 triton 使用的 driver 切成 Paddle 专用的 driver，确保其中使用的 API 都是 Paddle 的，在运行结束后再切回原来默认的，确保在 Paddle 环境下的 triton kernel 和在 torch 环境下的 triton kernel 都使用正确的 API，有极小的运行时开销

为了确保 FD 中两种场景都能正确执行，在 `enable_compat_on_triton_kernel` 里会根据环境是否安装 torch 来 dispatch 到上述两种逻辑

**这里本质问题是，PyTorch 和 Paddle 的 triton kernel 抢 triton 模块里一个 driver 的问题**，当然，实际上之前出现的问题还混合了在 enable compat 下，PyTorch 和 Paddle 抢 `sys.modules["torch.*"]` 的问题，不过现在在混合 PyTorch 场景下不再全局开，该问题规避掉了

## Usage or Command

后续所有 FD 里在 Paddle 下跑的 triton kernel 都应在 `@triton.jit` 上额外装饰 `@enable_compat_on_triton_kernel`，支持 PyTorch 和 Paddle 混用

```python
import triton
import torch
import paddle
from fastdeploy.model_executor.ops.triton_ops.triton_utils import (
    enable_compat_on_triton_kernel,
)

@triton.jit
def kernel_for_torch(x, y):
    ...

@enable_compat_on_triton_kernel
@triton.jit
def kernel_for_paddle(x, y):
    ...

kernel_for_torch(torch.tensor(1), torch.tensor(2))
kernel_for_paddle(paddle.tensor(1), paddle.tensor(2))
```

## Accuracy Tests

无

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [x] Provide accuracy results.
- [x] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

**Review by EmmonsCurse — APPROVED** [review]
LGTM～ Skip coverage check as it mainly relies on tests with torch.

**Review by K11OntheBoat — APPROVED** [review]
LGTM

**Comment by EmmonsCurse** [comment]
❌ Cherry-pick failed: Conflicts detected when cherry-picking to `release/2.5`. Please resolve manually.
